Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificates.kt

```diff
@@ -23,16 +25,23 @@ data class PersonCertificates(
     }
 
     // PersonOverview
-    val overviewCertificates: List<VerificationCertificate> by lazy {
-        // TODO .take(2)
-        dccWalletInfo?.verification?.certificates?.mapNotNull { certRef ->
-            certRef.buttonText
-            certificates.firstOrNull { it.qrCodeHash == certRef.certificateRef.barcodeData.toSHA256() }
+    val certificatesForOverviewScreen: List<VerificationCertificate> by lazy {
+        dccWalletInfo?.verification?.certificates.orEmpty().mapNotNull { certRef ->
+            certificates.firstOrNull { it.qrCodeHash == certRef.certificateRef.barcodeData.toSHA256() }?.let {
+                VerificationCertificate(
+                    certificate = it,
+                    buttonText = formatCCLText(certRef.buttonText, Locale.getDefault().language)
+                )
+            }
+        }.take(2).ifEmpty {
+            when (val cert = certificates.findFallbackDcc()) {
+                null -> emptyList()
+                else -> listOf(VerificationCertificate(cert))
+            }
         }
-
-        listOf(VerificationCertificate(certificate = certificates.findFallbackDcc()))
     }
 
+    // Obsolete, remove in cleanup PR
     val admissionState: AdmissionState? get() = certificates.determineAdmissionState()
```

Comment:
as it's not used anymore feel free to remove it now

Reply:
I thought about it, but I would rather have it removed in the cleanup PR because if I start with this, then I should also remove the `determineAdmissionState` function from `PersonCertificateExtensions`, and if I'm there, I should probably clean the rules, and so on 😄  I will add a `TODO` to the comment for better visibility.